src/Task.cc

```diff
@@ -721,6 +720,9 @@ void Task::on_syscall_exit_arch(int syscallno, const Registers& regs) {
           break;
         }
         case PTRACE_ARCH_PRCTL: {
+          if (tracee->arch() != x86_64) {
+            break;
```

Comment:
And then you won't need this.

Reply:
PTRACE_ARCH_PRCTL can be issues by an x86_64 tracer on an x86 tracee, which this is supposed to catch. I just moved this up from a bit below.

Reply:
Ah good point.